src/datatype.c

```diff
@@ -645,14 +645,7 @@ void jl_compute_field_offsets(jl_datatype_t *st)
         }
     }
     else {
-        // compute a conservative estimate of whether there could exist an instance of a subtype of this
-        for (i = 0; st->has_concrete_subtype && i < nfields - st->name->n_uninitialized; i++) {
-            jl_value_t *fld = jl_svecref(st->types, i);
-            if (fld == jl_bottom_type)
-                st->has_concrete_subtype = 0;
-            else
-                st->has_concrete_subtype = !jl_is_datatype(fld) || ((jl_datatype_t *)fld)->has_concrete_subtype;
-        }
+        jl_compute_has_concrete_subtype(st);
```

Comment:
It seems to still be necessary to call this here, for types such as `A57757{Int,Union{}}`, otherwise `has_concrete_subtype` was still set to `1`.